go/host/batchmanager/batch_manager.go

```diff
@@ -96,9 +100,14 @@ func (b *BatchManager) GetBatches(batchRequest *common.BatchRequest) ([]*common.
 			break
 		}
 		batchesToSend = append(batchesToSend, currentBatch)
-		currentBatch, err = b.db.GetBatch(currentBatch.Header.ParentHash)
+		hashToLookFor := currentBatch.Header.ParentHash
+		currentBatch, err = b.db.GetBatch(hashToLookFor)
 		if err != nil {
-			return nil, fmt.Errorf("could not retrieve batch header. Cause: %w", err)
+			currentBatch, err = enclaveClient.GetBatch(hashToLookFor)
```

Comment:
Is it important to check for the error type here ?

Reply:
Like check for network errors since we are using grpc underlying? I guess I can add a todo to do it with retry logic or something smarter

Reply:
I should be clearer, ehehe. 
If the `currentBatch, err = b.db.GetBatch(hashToLookFor)` fails, the error gets ignored and immediately is fetched from the enclave.
Worth checking the error I guess and only fetching from the enclave if it's an `errutil.ErrNotFound` ?